### DISCUSSION TITLE
API route for preview not successfully capturing query params

### DISCUSSION THREAD

**Discussion by readeral** [vercel/next.js#15419] [Help]
# Bug report

Having imitated https://github.com/vercel/next.js/tree/canary/examples/cms-sanity to set up preview mode for a Sanity/NextJS project, I'm unable to load a page in preview mode in production as NextJS isn't picking up the secret value or slug in its request.query object.

## To Reproduce

The URL is generated through a function on my Sanity project:
```
const previewSecret = '032ac2eecbe611ea87d00242ac130003'; // Copy the string you used for SANITY_PREVIEW_SECRET
const projectUrl =
  process.env.NODE_ENV === 'development'
    ? 'http://localhost:3000'
    : 'https://stkildapc.org';


export default function resolveProductionUrl(document) {
  return `${projectUrl}/api/preview?secret=${previewSecret}&slug=${document.slug.current}`;
}
```

Which accurately produces the URL https://www.stkildapc.org/api/preview?secret=032ac2eecbe611ea87d00242ac130003&slug=draft
Clicking out from Sanity, after opening in a new tab this URL flashes momentarily (confirmed by taking a screenshot!) but results in request.query being empty, and returning 401 Invalid Token.
See the API route code here: https://github.com/NewFrontDoor/stkildapc-nextjs/blob/master/pages/api/preview.js

I've confirmed that the API route isn't correctly picking up the URL query parameters by logging it out to the console and checking the realtime logs for the API function in Vercel dashboard, which returns an empty object.

The result from my 3 console logs on lines 15-17 of the repo above return the following:
{
  ...
  query: [Getter/Setter],
  body: '',
  [Symbol(kCapture)]: false
}
{}
032ac2eecbe611ea87d00242ac130003

These functions work correctly when I'm running NextJS on localhost:3000, but fail once I'm trying to reach the preview mode from a deployed NextJS site. The produced URL also works correctly if just visiting directly.

## Expected behavior

When clicking 'Open Preview' in Sanity, and navigating to my preview API route, that the preview page would correctly load.

## Screenshots

<img width="1086" alt="Screen Shot 2020-07-22 at 10 24 03 pm" src="https://user-images.githubusercontent.com/15904136/88240767-7913f580-cccb-11ea-82aa-6d4b3135dcfc.png">
<img width="1062" alt="Screen Shot 2020-07-22 at 10 24 37 pm" src="https://user-images.githubusercontent.com/15904136/88240795-8af59880-cccb-11ea-9644-b330bde9e4ae.png">

## System information

- macOS
- Tested in chrome & safari
- Version of Next.js: 9.4.4
- Version of Node.js: 14.5.0

## Additional context

N.b. I've included the token in my issue - obviously will be changing it after this is resolved...

**Comment by AlexandraKlein** [comment]
I'm having the same issue. On prod, query params are undefined. Returns as intended in development.

```javascript
export default async function preview(req, res) {
   const { slug, id } = req.query;
   console.log({ slug,  id }); // Both undefined in production.
}
```

**Comment by iisalazar** [comment]
Any update on this bug? I'm facing the same issue here

**Reply by iisalazar** [reply]
nevermind I found the issue. There was a typo on the NEXT_PUBLIC_SANITY_PROJECT_ID environment variable in ```.env.local```

**Comment by koraysels** [comment]
I am facing this bug now... in development it has the quey params but not in production ...
